### ISSUE TITLE
Palette Storage

### ISSUE THREAD

**Issue by SteveTheTechie** [bgrins/spectrum#278]
One of the things was getting some negative feedback from my users about was the ability to modify the palette.  I mentioned to them that they could store selected colors into the palette, only to find out that not everybody has local storage enabled or even necessarily knows what it is.

Because of this, I started digging into the local storage stuff in spectrum.   I created a variation of spectrum that my users are running that does the following:
1.  It sets a haveLocalStorage variable during initialization.  One of the things I was discovering was that, while the try / catch for the local storage would certainly handle the case of local storage not being available/enabled, the exception handling was adding overhead for the 15 instances of spectrum I am creating that was noticeably slowing things down.  Therefore, what I did was to implement a single check (with the exception handling) for whether local storage is available, and just reference that status variable later.   Yeah, it is not as rigorous as doing the exception handling on every local storage access, but I tend to think the risk is very low and reasonable.  It does improve performance a bit.
2.  I decided I still want to enable my users to save the selection palette even if they do not have local storage available/enabled.  Because of that, I decided to add 3 lines of code to implement cookie storage as a fall back if local storage is not available/enabled.  This way, my users are more likely to be able to actually make use of the selection palette storage feature regardless of whether they have or know how to enable local storage.

Please let me know if you would be interested in me doing a pull request for either of the above.  As noted above, my variation is running as intended.  (For 10,000+ Toastmasters public speaking club websites around the world.)

**Comment by bgrins** [comment]
As for 1, I think it'd be fine to use a single variable for checking if it's available.  When actually setting a value into storage we should still keep a try/catch because there can be other exceptions (like if the localStorage space is filled up).

For 2, if it's just a couple of lines I think it'd be fine to fall back to cookie storage.  We should probably put that behind an option in case users don't want to have cookies set.

**Comment by SteveTheTechie** [comment]
Good point on 1.  In my case, the performance hit I was noticing was during the initialization which would be for the initial read from storage, not the storing of a value.  I think that typically storing the palette would happen one at a time even if there are a lot of instances.

On 2, I was thinking of an option like "allowCookies" or similar, with a default value of true.  It would just work as a fall back in the catch of the try/catch, for instance.
